src/main/kotlin/com/openlattice/shuttle/IntegrationService.kt

```diff
@@ -0,0 +1,349 @@
+        entitySetManager.deleteEntitySet(integration.logEntitySetId.get())
+    }
+
+    private fun checkIntegrationJobExists(jobId: UUID) {
+        checkState(integrationJobs.containsKey(jobId), "Job Id $jobId is not assigned to an existing integration job")
+    }
+
+    private fun checkIntegrationExists(integrationName: String) {
+        checkState(integrations.containsKey(integrationName), "Integration with name $integrationName does not exist.")
+    }
+
+    private fun getIdToken(creds: Properties): String {
+        return authAPI
+                .login(creds.getProperty("email"), creds.getProperty("password"), creds.getProperty("realm"))
+                .setScope(creds.getProperty("scope"))
+                .setAudience(creds.getProperty("audience"))
+                .execute()
+                .idToken
+    }
+
+    private fun getSrcDataSource(source: Properties): HikariDataSource {
+        return HikariDataSource(HikariConfig(source))
+    }
+
+    private fun generateDestinationsMap(integration: Integration, missionParameters: MissionParameters, dataIntegrationApi: DataIntegrationApi): Map<StorageDestination, IntegrationDestination> {
+        val s3BucketUrl = integration.s3bucket
+        val dstDataSource = HikariDataSource(HikariConfig(missionParameters.postgres.config))
+        integration.maxConnections.ifPresent { dstDataSource.maximumPoolSize = it }
+
+        val pgDestination = PostgresDestination(
+                entitySets.mapKeys { it.value.id },
+                entityTypes,
+                propertyTypes.mapKeys { it.value.id },
+                dstDataSource
+        )
+
+        if (s3BucketUrl.isBlank()) {
+            return mapOf(StorageDestination.POSTGRES to pgDestination)
+        }
+        val s3Api = Retrofit.Builder()
+                .baseUrl(s3BucketUrl)
+                .addConverterFactory(RhizomeByteConverterFactory())
+                .addConverterFactory(RhizomeJacksonConverterFactory(ObjectMappers.getJsonMapper()))
+                .addCallAdapterFactory(RhizomeCallAdapterFactory())
+                .client(RetrofitBuilders.okHttpClient().build())
+                .build().create(S3Api::class.java)
+        val s3Destination = PostgresS3Destination(pgDestination, s3Api, dataIntegrationApi)
+        return mapOf(StorageDestination.POSTGRES to pgDestination, StorageDestination.S3 to s3Destination)
+    }
+
+    private fun createLogEntitySet(integrationName: String, integration: Integration): UUID {
+        val name = buildLogEntitySetName(integrationName)
+        val contacts = integration.contacts
+        val description = buildLogEntitySetDescription(integration.flightPlanParameters)
+        val logEntitySet = EntitySet(
+                logEntityType.id,
+                name,
+                name,
+                Optional.of(description),
+                contacts,
+                Optional.empty(),
+                IdConstants.GLOBAL_ORGANIZATION_ID.id,
```

Comment:
sooo I feel like the right thing to do here is to force an `Integration` object to have an `organizationId` and use that id here -- for changeover purposes we could give it a default value of the global org, but ultimately integrations should probably be tied to a particular organization / we should push towards not just putting all new entity sets in the global org

Reply:
you are very much right. I will add this.